libraries/ms-common/src/x86/uart.c

```diff
@@ -1,13 +1,90 @@
 #include "uart.h"
+#include "log.h"
+#include "x86_socket.h"
+#include <stdlib.h>
+#include <string.h>
+
+typedef struct {
+  UARTPort port;
+  X86SocketThread thread;
+  UARTStorage storage;
```

Comment:
The storage provided by the user is required to persist per the API. Why not store a pointer to it instead of copying data in?

Reply:
the struct now contains a pointer to a UARTStorage instead